File: app/src/main/java/ruan/com/Net/HttpManager/Interface/HttpCallback.java
package ruan.com.Net.HttpManager.Interface;

import ruan.com.Net.HttpManager.HttpResponse;
import ruan.com.retrofit2.BaseObject;

/**
 * Created by 19820 on 2018/1/8.
 */

public interface HttpCallback{


    interface Response{
        /**
         * 请求成功
         * @param requestCode       请求标识
         * @param response        请求返回的对象
         */
        void onSuccess(int requestCode , HttpResponse response);

        /**
         * 请求失败
         * @param requestCode       请求标识
         * @param throwable         请求失败的对象
         */
        void onError(int requestCode , Throwable throwable);

    }
}

File: app/src/main/java/ruan/com/Net/HttpManager/Interface/HttpCallback.kt
package ruan.com.Net.HttpManager.Interface

import ruan.com.Net.HttpManager.HttpResponse

/**
 * Created by Administrator on 2018/5/4.
 */
interface HttpCallback {

    interface Response{
        /**
         * 返回成功
         *
         * @param  requestCode  返回码
         * @param  response     返回对象
         */
        fun onSuccess(requestCode: Int, response: HttpResponse?)

        /**
         * 返回失败
         *
         * @param  requestCode  返回码
         * @param  throwable    返回错误
         */
        fun onError(requestCode: Int , throwable: Throwable?)

    }

}